app/controllers/products_controller.rb

```diff
@@ -9,23 +9,43 @@ def index
   def show ; end
 
   def new
-    @product = Product.new
+    if session[:user_id] != nil 
+      @product = Product.new
+    else
```

Comment:
I understand the logic here- but is it enough to know that session[:user_id] is not nil? I mean, session could be overwritten by any bad actor. Do we want additional logic to determine whether the session[:user_id] value corresponds with an authenticated user? or can we assume that this already happens?